### PR TITLE
Unify Expression and IntegralIR with a common base

### PR DESCRIPTION
**Pull request by jorgensd**
Find commonground between IntegralIR and ExpressionIR.
Various clean-ups of extra parameters (such as points in expressionIR, as they are inside the integrand map key).

### CODE REVIEW

**Review by mscroggs — APPROVED** [review]

File: ffcx/codegeneration/definitions.py [hunk 1]
```diff
@@ -50,16 +50,15 @@ def create_dof_index(tabledata, dof_index_symbol):
...
-        self.integral_type = ir.integral_type
-        self.entitytype = ir.entitytype
+        self.integral_type = integral_type
+        self.entity_type = entity_type
         self.access = access
-        self.symbols = access.symbols
         self.options = options
 
-        self.ir = ir
+        # self.ir = ir
```
Comment (mscroggs) [inline ffcx/codegeneration/definitions.py:62]:
```suggestion
```

File: ffcx/ir/representation.py [hunk 2]
```diff
@@ -62,13 +63,51 @@ class QuadratureIR(typing.NamedTuple):
     weights: npt.NDArray[np.float64]
 
 
-class IntegralIR(typing.NamedTuple):
-    """Intermediate representation of an integral."""
+class ElementIR(typing.NamedTuple):
```
Comment (mscroggs) [inline ffcx/ir/representation.py:66]: This can probably be removed

File: ffcx/ir/representation.py [hunk 3]
```diff
@@ -62,13 +63,51 @@ class QuadratureIR(typing.NamedTuple):
...
+    name: str
+    signature: str
+    num_global_support_dofs: int
+    num_element_support_dofs: int
+    entity_dofs: list[list[list[int]]]
+    entity_closure_dofs: list[list[list[int]]]
+    num_entity_closure_dofs: list[list[int]]
+    num_sub_dofmaps: int
+    sub_dofmaps: list[str]
+    block_size: int
```
Comment (jorgensd) [inline ffcx/ir/representation.py:103]:
```suggestion
```

File: .github/workflows/dolfinx-tests.yml [hunk 4]
```diff
@@ -51,7 +51,7 @@ jobs:
         with:
           path: ./dolfinx
           repository: FEniCS/dolfinx
-          ref: main
+          ref: dokken/unify_irs
```
Comment (jorgensd) [inline .github/workflows/dolfinx-tests.yml:54]:
```suggestion
          ref: main
```

**Comment by jhale** [comment]
This was merged without the DOLFINx Integration tests passing - easy to miss this, perhaps it should be on the default set?

**Comment by jorgensd** [comment]
> This was merged without the DOLFINx Integration tests passing - easy to miss this, perhaps it should be on the default set?

That is just because of the branch reset to main. Has to happen here prior to dolfinx gets merged

**Pull request merged**